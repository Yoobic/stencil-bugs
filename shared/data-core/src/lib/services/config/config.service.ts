import { Injectable, Injector } from '@angular/core';
import { LocalStorage, ConfigConstants, CoreConfig } from '@shared/common';
import { Translate } from '@shared/translate';

@Injectable()
export class Config {
    private static PROD_URL = 'https://api3.yoobic.com/'; //https://yoobic-loopback-prod-v3.herokuapp.com/'; // 'https://api.yoobic.com/'
    //private static CHINA_PROD_URL = 'https://china.yoobic.com/'; //https://yoobic-loopback-prod-v3.herokuapp.com/'; // 'https://api.yoobic.com/'
    private static STAGING_URL = 'https://yoobic-loopback-staging-v3.herokuapp.com/';
    private static DEMO_URL = 'https://yoobic-loopback-demo.herokuapp.com/';
    private static DEV_URL = 'https://yoobic-loopback-dev-v3.herokuapp.com/';
    private static DEV1_URL = 'https://yoobic-loopback-dev1-v3.herokuapp.com/';
    private static LOCALHOST_URL = 'https://localhost:3000/';
    private static TESTPEN_URL = 'https://testpen.yoobic.com/';
    private static TRIAL_URL = 'https://yoobic-loopback-trial.herokuapp.com/';
    private static E2E_URL = 'https://yoobic-loopback-e2e.herokuapp.com/';

    private static IMAGE_URL = 'https://upload.yoobic.com/api/'; // 'https://images.yoobic.com/api/ImageContainers/cloudinary/upload'; //'https://192.168.1.88:3000/api/ImageContainers/cloudinary/upload'//'http://localhost:3000/api/ImageContainers/cloudinary/upload'; //

    private static MAPPING_PROD_URL = 'https://etl.yoobic.com/flows/mapping/upload_loopback/yoobic-production-v3/';
    private static MAPPING_DEMO = 'https://etl.yoobic.com/flows/mapping/upload_loopback/yoobic-demo/';
    private static MAPPING_STAGING_URL = 'https://etl.yoobic.com/flows/mapping/upload_loopback/yoobic-staging/';
    private static MAPPING_DEV_URL = 'https://etl.yoobic.com/flows/mapping/upload_loopback/yoobic-dev-v3/';
    private static MAPPING_DEV1_URL = 'https://etl.yoobic.com/flows/mapping/upload_loopback/yoobic-dev0/';

    private static ZAPIER_INSTAGRAM_PROD_URL = 'https://etl.yoobic.com/flows/zapier_instagram_start';
    protected translate: Translate;
    private server: string;
    private configServer;

    constructor(protected localStorage: LocalStorage, configConstants: ConfigConstants, protected coreConfig: CoreConfig, protected injector: Injector) {
        this.configServer = configConstants.server;
        this.translate = this.injector.get<Translate>(Translate);
    }

    public get servers() {
        let servers = [
            { _id: 'prod', name: 'Production', url: Config.PROD_URL },
            { _id: 'demo', name: 'Demo', url: Config.DEMO_URL },
            { _id: 'staging', name: 'Staging', url: Config.STAGING_URL },
            { _id: 'dev', name: 'Development', url: Config.DEV_URL },
            { _id: 'dev1', name: 'Development 1', url: Config.DEV1_URL },
            { _id: 'trial', name: 'Trial', url: Config.TRIAL_URL },
            { _id: 'e2e', name: 'E2E', url: Config.E2E_URL }

        ];
        if (!this.coreConfig.isUniversal() || location.hostname === 'localhost') {
            servers.push({ _id: 'localhost', name: 'Localhost', url: Config.LOCALHOST_URL });
            servers.push({ _id: 'testpen', name: 'Testpen', url: Config.TESTPEN_URL });
        }
        if (this.configServer) {
            servers = [this.configServer];
        }

        if (this.isTestpen) {
            servers = [{ _id: 'testpen', name: 'Testpen', url: Config.TESTPEN_URL }];
        }
        return servers;
    }

    public get serverUrl() {
        if (this.isTestpen) {
            return Config.TESTPEN_URL;
        }
        let defaultServer;
        if (this.server && this.server !== '') {
            defaultServer = this.server;
        } else {
            // if (this.translate.isCurrentLanguageChinese()) {
            //     defaultServer = Config.CHINA_PROD_URL;
            // } else {
            defaultServer = Config.PROD_URL; // this.mode === 'prod' ? Config.PROD_URL : Config.DEV_URL;}
            //}
        }
        let configServer = this.configServer && this.configServer.url ? this.configServer.url : undefined;
        this.server = this.localStorage.get('SERVER') || configServer || defaultServer;
        return this.server;
    }

    public get serverName() {
        let server = this.servers.filter(s => s.url === this.serverUrl);
        if (server && server.length === 1) {
            return server[0].name;
        }
        return this.translate.get('CUSTOM');
    }

    public set serverUrl(url) {
        this.localStorage.set('SERVER', url);
    }

    public get apiUrl() {
        return this.serverUrl + 'api/';
    }

    public get uploadUrl() {
        return (this.localStorage.get('SERVER_IMAGE') || Config.IMAGE_URL) + 'ImageContainers/cloudinary/upload';
    }

    public get uploadProxyUrl() {
        return (this.localStorage.get('SERVER_IMAGE') || Config.IMAGE_URL) + 'cloudinary/uploadProxyImage';
    }

    public get mappingUrl() {
        let url = this.serverUrl;

        let retVal: string;
        switch (url) {
            case Config.DEMO_URL:
                retVal = Config.MAPPING_DEMO;
                break;
            case Config.STAGING_URL:
                retVal = Config.MAPPING_STAGING_URL;
                break;
            case Config.DEV1_URL:
                retVal = Config.MAPPING_DEV1_URL;
                break;
            case Config.PROD_URL:
                retVal = Config.MAPPING_PROD_URL;
                break;
            case Config.DEV_URL:
            default:
                retVal = Config.MAPPING_DEV_URL;
                break;
        }
        return this.localStorage.get('SERVER_MAPPING') || retVal;
    }

    public get zapierInstagramUrl() {
        let url = this.serverUrl;
        switch (url) {
            case Config.PROD_URL:
                return Config.ZAPIER_INSTAGRAM_PROD_URL;
            default:
                return null;
        }
    }

    public get isTestpen() {
        return !this.coreConfig.isUniversal() && (location.hostname === 'testpen-dashboard.yoobic.com' || location.hostname === 'testpen-mobile.yoobic.com' || this.server === Config.TESTPEN_URL);
    }

    public get isE2E() {
        return this.serverUrl === Config.E2E_URL;
    }
}
