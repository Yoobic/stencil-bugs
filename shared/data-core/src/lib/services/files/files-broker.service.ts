import { Injectable } from '@angular/core';

import { Files } from '../files/files.service';
import { Broker, ResponseObject } from '../broker/broker.service';
import { Folder } from '../../interfaces/folder/folder.interface';
import { Dashboard } from '../dashboard/dashboard.service';

import { Observable, Observer } from 'rxjs';

@Injectable()
export class FilesBroker {
    constructor(private dashboard: Dashboard, private broker: Broker, protected files: Files) { }

    getFilesTransform() {
        return (res: ResponseObject) => {
            if (res && res.data && res.data.map) {
                res.data.forEach((f) => this.updateFileIcon(f));
            }
            return res;
        };
    }

    getFilesFoldersTransformAsync() {
        return (res: ResponseObject, search, filters, start, pageSize) => {
            return new Observable<ResponseObject>((observer: Observer<ResponseObject>) => {
                if (res && res.data && res.data.length > 0) {
                    res.data.forEach(f => {
                        delete f.items;
                        if (f.fftype === 'file') {
                            this.updateFileIcon(f);
                        }
                    });
                    let ids = res.data.filter(f => f.fftype === 'folder').map((folder: Folder) => folder._id);
                    this.dashboard.getFolderFolderStat(ids).subscribe((stats) => {
                        res.data.forEach((folder: Folder) => {
                            let stat = stats.find(s => s._id === folder._id);
                            let value = stat ? stat.folders || 0 : 0;
                            folder.stats = [{ title: 'FOLDERS', color: 'dark', value }];
                            (<any>folder).hasChildren = value > 0;
                        });
                        this.dashboard.getFolderFileStat(ids, false).subscribe((fileStats) => {
                            res.data.forEach((folder: Folder) => {
                                let stat = fileStats.find(s => s._id === folder._id);
                                folder.stats.push({ title: 'FILES', color: 'dark', value: stat ? stat.files || 0 : 0 });
                            });
                            observer.next({ count: res.count, data: <any>res.data });
                            observer.complete();
                        });
                    });
                } else {
                    observer.next({ count: 0, data: [] });
                    observer.complete();
                }
            });
        };
    }

    cleanupFolder(folderId) {
        return this.broker.deleteAll('files', { where: { 'folder': { 'eq': folderId } } });
    }

    updateFileIcon(f) {
        if (this.files.isImage(f)) {
            f.imgSrc = f._downloadURL;
        } else if (this.files.isVideo(f)) {
            f.imgSrc = this.files.getVideoPoster(f._downloadURL);
        } else {
            f.icon = this.files.getIcon(f);
        }
    }

}
