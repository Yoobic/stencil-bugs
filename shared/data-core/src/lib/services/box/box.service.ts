import { Injectable } from '@angular/core';
import { Config } from '../config/config.service';
import { Requestor } from '../requestor/requestor.service';
import { Observable } from 'rxjs';
import { CoreConfig } from '@shared/common';
import { map } from 'rxjs/operators';

@Injectable()
export class Box {

    constructor(private config: Config, private rq: Requestor, private coreConfig: CoreConfig) { }

    upload(documentUrl: string): Observable<any> {
        let url = this.config.apiUrl + 'businesslogic/uploadToBox';
        return this.rq.post(url, { params: { documentUrl } }).pipe(map(retVal => {
            if (retVal && retVal.data) {
                return retVal.data.id;
            }
            return null;
        }));
    }

    createViewingSession(boxId: string): Observable<any> {
        let url = this.config.apiUrl + 'businesslogic/createViewingSessionBox';
        return this.rq.post(url, { params: { boxId } }).pipe(map(retVal => {
            if (retVal && retVal.data) {
                return retVal.data;
            }
            return null;
        }));
    }

    getViewingContent(contentUrl: string): Observable<any> {
        let url = this.config.apiUrl + 'businesslogic/getViewingContentBox';
        return this.rq.post(url, { params: { contentUrl } }).pipe(map(retVal => {
            if (retVal && retVal.data) {
                let content = retVal.data;
                let headIndex = content.indexOf('<head>') + 6;
                let overwriteStyle = '<style type="text/css">';
                overwriteStyle += '.controls-center .scroll-previous-btn, .controls-center .scroll-next-btn, .controls-right .zoom-in-btn, .controls-right .zoom-out-btn, .controls-right {';
                overwriteStyle += '  display: initial !important;';
                overwriteStyle += '}';
                overwriteStyle += '.fullscreen-btn {';
                overwriteStyle += '  display: none !important;';
                overwriteStyle += '}';
                overwriteStyle += '</style>';
                let protocol = this.coreConfig.getProtocol();
                content = [content.slice(0, headIndex), overwriteStyle, content.slice(headIndex)].join('');
                content = content.replace(new RegExp('href="//', 'g'), 'href="' + protocol + '//');
                content = content.replace(new RegExp('src="//', 'g'), 'src="' + protocol + '//');
                return content;
            }
            return null;
        }));
    }

}
