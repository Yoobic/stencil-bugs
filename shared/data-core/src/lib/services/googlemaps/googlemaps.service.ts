/* tslint:disable:variable-name */
import { Injectable, Injector } from '@angular/core';
import { Config } from '../config/config.service';
import { Requestor } from '../requestor/requestor.service';

import { Smartloc } from '../smartloc/smartloc.service';
import { Position } from '@shared/common';
import { Translate } from '@shared/translate';
import { IAutocompleteInput, IAutocompletePrediction, IAutocompleteResponse, IBounds, IGeocodeInput, IGeocodeResult, IGeocodeResponse, INearbySearchInput, INearbySearchResponse, IReverseGeocodeInput, IReverseGeocodeResponse, IResolvedAddressLocationInput, IResolvedAddressLocationResult } from '../../interfaces/googlemaps/googlemaps.interface';

import { concatMap, map } from 'rxjs/operators';
import { Observable, from, combineLatest } from 'rxjs';

function normalizeBounds(b: string | IBounds): string {
    return b ? (typeof b === 'string' ? b : b.southwest.lat + ',' + b.southwest.lng + '|' + b.northeast.lat + ',' + b.northeast.lng) : '';
}

function normalizeLocation(p: string | Position): string {
    return p ? (typeof p === 'string' ? p : p.toStringLoc()) : '';
}

function pipeJoin(p: string | string[]): string {
    return p ? (typeof p === 'string' ? p : p.join('|')) : '';
}

function normalizeObservable<T>(o: Observable<T> | Promise<T> | T): Observable<T> {
    return o instanceof Observable ? o : from(Promise.resolve(o));
}

@Injectable()
export class Googlemaps {
    private translate: Translate;
    constructor(private rq: Requestor, private geo: Smartloc, private config: Config, protected injector: Injector) {
        this.translate = this.injector.get<Translate>(Translate);
    }

    public autocompleteFromSync(
        input: string,
        location: string | Position,
        offset: number = input.length,
        radius = 100000,
        language: string = this.translate.getCurrentLanguage(),
        types?: string,
        components?: string,
        query = false
    ): Observable<IAutocompletePrediction[]> {
        return this._autocomplete({
            input,
            offset,
            location: normalizeLocation(location),
            radius,
            language,
            types,
            components
        }, query);
    }

    public autocompleteLocationObserver(
        input: string,
        offset: number = input.length,
        location: Observable<string | Position> | Promise<string | Position> | string | Position = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        radius = 1000,
        language: string = this.translate.getCurrentLanguage(),
        types?: string,
        components?: string,
        query = false
    ): Observable<IAutocompletePrediction[]> {
        return normalizeObservable(location).pipe(
            map((loc): IAutocompleteInput => {
                return {
                    input,
                    offset,
                    location: normalizeLocation(loc),
                    radius,
                    language,
                    types,
                    components
                };
            }),
            concatMap((i) => this._autocomplete(i, query))
        );
    }

    public autocompleteMultiObserver(
        inputO: Observable<string>,
        offsetO: Observable<number> = inputO.pipe(map(s => s.length)),
        locationO: Observable<string | Position> = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        radius = 1000,
        language: string = this.translate.getCurrentLanguage(),
        types?: string,
        components?: string,
        query = false
    ): Observable<IAutocompletePrediction[]> {
        return combineLatest(inputO, offsetO, locationO)
            .pipe(concatMap((i: any) => this._autocomplete(i, query)));
    }

    public nearbySearchFromSync(
        location: string | Position,
        radius = 1000,
        keyword?: string,
        language: string = this.translate.getCurrentLanguage(),
        name?: string | string[],
        minprice?: number,
        maxprice?: number,
        opennow?: boolean,
        rankby?: string,
        type?: string,
        types?: string | string[]
    ): Observable<INearbySearchResponse> {
        return this._nearbySearch({
            location: normalizeLocation(location),
            radius,
            keyword,
            language,
            name: pipeJoin(name),
            minprice,
            maxprice,
            opennow,
            rankby,
            type,
            types: pipeJoin(types)
        });
    }

    public nearbySearchNextResults(
        next_page_token: string
    ): Observable<INearbySearchResponse> {
        return this._nextSearchResults(next_page_token);
    }

    public nearbySearchLocationObserver(
        location: Observable<string | Position> | Promise<string | Position> | string | Position = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        radius = 1000,
        keyword?: string,
        language: string = this.translate.getCurrentLanguage(),
        name?: string | string[],
        minprice?: number,
        maxprice?: number,
        opennow?: boolean,
        rankby?: string,
        type?: string,
        types?: string | string[]
    ): Observable<INearbySearchResponse> {
        return normalizeObservable(location).pipe(
            map((loc): INearbySearchInput => {
                return {
                    location: normalizeLocation(loc),
                    radius,
                    keyword,
                    language,
                    name: pipeJoin(name),
                    minprice,
                    maxprice,
                    opennow,
                    rankby,
                    type,
                    types: pipeJoin(types)
                };
            }),
            concatMap((input) => this._nearbySearch(input))
        );
    }

    public nearbySearchMultiObserver(
        locationO: Observable<string | Position> = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        keywordO?: Observable<string>,
        radius = 1000,
        language: string = this.translate.getCurrentLanguage(),
        nameO?: Observable<string | string[]>,
        minprice?: number,
        maxprice?: number,
        opennow?: boolean,
        rankby?: string,
        type?: string,
        types?: string | string[]
    ): Observable<INearbySearchResponse> {
        return combineLatest(keywordO, nameO, locationO)
            .pipe(concatMap((i: any) => this._nearbySearch(i)));
    }

    public geocodeFromSync(
        address?: string,
        components?: string | string[],
        bounds?: string | IBounds,
        language: string = this.translate.getCurrentLanguage(),
        region?: string
    ): Observable<IGeocodeResult[]> {
        return this._geocode({
            address,
            components: pipeJoin(components),
            bounds: normalizeBounds(bounds),
            language,
            region
        });
    }

    public geocodeAddressObserver(
        address: Observable<string>,
        components?: string | string[],
        bounds?: string | IBounds,
        language: string = this.translate.getCurrentLanguage(),
        region?: string
    ): Observable<IGeocodeResult[]> {
        return address.pipe(
            map((addr): IGeocodeInput => {
                return {
                    address: addr,
                    components: pipeJoin(components),
                    bounds: normalizeBounds(bounds),
                    language,
                    region
                };
            }),
            concatMap((i) => this._geocode(i))
        );
    }

    public reverseGeocodeFromSyncLocation(
        latlng: string | Position,
        language: string = this.translate.getCurrentLanguage(),
        result_type?: string,
        location_type?: string
    ): Observable<IGeocodeResult[]> {
        return this._reverseGeocode({
            latlng: normalizeLocation(latlng),
            language,
            result_type,
            location_type
        });
    }

    public reverseGeocodeLatLngObserver(
        latlng: Observable<string | Position> | Promise<string | Position> | string | Position = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        language: string = this.translate.getCurrentLanguage(),
        result_type?: string,
        location_type?: string
    ): Observable<IGeocodeResult[]> {
        return normalizeObservable(latlng).pipe(
            map((loc): IReverseGeocodeInput => {
                return {
                    latlng: normalizeLocation(loc),
                    language,
                    result_type,
                    location_type
                };
            }),
            concatMap((i) => this._reverseGeocode(i))
        );
    }

    public reverseGeocodeFromSyncPlaceID(
        place_id: string,
        language: string = this.translate.getCurrentLanguage(),
        result_type?: string,
        location_type?: string
    ): Observable<IGeocodeResult> {
        return this._reverseGeocode({
            place_id,
            language,
            result_type,
            location_type
        }).pipe(map(results => results[0]));
    }

    public reverseGeocodePlaceIDObserver(
        place_idO: Observable<string>,
        language: string = this.translate.getCurrentLanguage(),
        result_type?: string,
        location_type?: string
    ): Observable<IGeocodeResult> {
        return place_idO.pipe(
            map((place_id): IReverseGeocodeInput => {
                return {
                    place_id,
                    language,
                    result_type,
                    location_type
                };
            }),
            concatMap((i) => this._reverseGeocode(i)),
            map(results => results[0])
        );
    }

    public placePredictionsLocationObserver(
        input?: string,
        location: Observable<string | Position> | Promise<string | Position> | string | Position = from(this.geo.getLocation().then(pos => pos.toStringLoc())),
        language: string = this.translate.getCurrentLanguage(),
        offset: number = input.length,
        radius?: number,
        types?: string,
        components?: string,
        address?: string,
        search?: string,
        type?: string,
        limit?: number,
        nearbyRadius?: number,
        skipName?: boolean,
        debug?: boolean
    ): Observable<IGeocodeResult[]> {
        return normalizeObservable(location).pipe(
            map((loc): IAutocompleteInput | IResolvedAddressLocationInput => {
                return {
                    input,
                    offset,
                    location: normalizeLocation(loc),
                    radius,
                    language,
                    types,
                    components,
                    address,
                    search,
                    type,
                    limit,
                    nearbyRadius,
                    skipName,
                    debug
                };
            }),
            concatMap((i) => this._placePredictions(i))
        );
    }
    public placePredictionsFromSync(
        location: string | Position,
        input?: string,
        language: string = this.translate.getCurrentLanguage(),
        offset: number = input.length,
        radius?: number,
        types?: string,
        components?: string,
        address?: string,
        search?: string,
        type?: string,
        limit?: number,
        nearbyRadius?: number,
        skipName?: boolean,
        debug?: boolean
    ): Observable<IGeocodeResult[]> {
        return this._placePredictions({
            input,
            offset,
            location: normalizeLocation(location),
            radius,
            language,
            types,
            components,
            address,
            search,
            type,
            limit,
            nearbyRadius,
            skipName,
            debug
        });
    }

    public nearbyPlaceCodesFromSync(
        location: string | Position,
        radius?: number,
        keyword?: string,
        language: string = this.translate.getCurrentLanguage(),
        name?: string | string[],
        minprice?: number,
        maxprice?: number,
        opennow?: boolean,
        rankby?: string,
        types?: string | string[],
        address?: string,
        search?: string,
        type?: string,
        limit?: number,
        nearbyRadius?: number,
        skipName?: boolean,
        debug?: boolean
    ): Observable<IGeocodeResult[]> {
        return this._nearbyPlaceCodes({
            location: normalizeLocation(location),
            radius,
            keyword,
            language,
            name,
            minprice,
            maxprice,
            opennow,
            rankby,
            types,
            address,
            search,
            type,
            limit,
            nearbyRadius,
            skipName,
            debug
        });
    }

    public resolveAddressLocation(
        address: string,
        language: string = this.translate.getCurrentLanguage(),
        search?: string,
        type?: string,
        limit?: number,
        nearbyRadius?: number,
        skipName?: boolean,
        debug?: boolean
    ): Observable<IResolvedAddressLocationResult> {
        return this._resolveAddressLocation({
            address,
            language,
            search,
            type,
            limit,
            nearbyRadius,
            skipName,
            debug
        });
    }

    private get apiUrl() {
        return this.config.apiUrl + 'GoogleMaps/';
    }

    private _autocomplete(input: IAutocompleteInput, query: boolean): Observable<IAutocompletePrediction[]> {
        return this.rq.post(`${this.apiUrl}place/autocomplete?query=${query}`, input).pipe(map((res: IAutocompleteResponse) => {
            return res.predictions;
        }));
    }

    private _placePredictions(input: IResolvedAddressLocationInput | IAutocompleteInput): Observable<IGeocodeResult[]> {
        return this.rq.post(this.apiUrl + 'place/predictions', input);
    }

    private _nearbyPlaceCodes(input: IResolvedAddressLocationInput | INearbySearchInput): Observable<IGeocodeResult[]> {
        return this.rq.post(this.apiUrl + 'place/nearby-place-codes', input);
    }

    private _resolveAddressLocation(input: IResolvedAddressLocationInput): Observable<IResolvedAddressLocationResult> {
        return this.rq.post(this.apiUrl + 'resolve-address-location', input);
    }

    private _nearbySearch(input: INearbySearchInput): Observable<INearbySearchResponse> {
        return this.rq.post(this.apiUrl + 'place/nearby', input);
    }

    private _nextSearchResults(next_page_token: string): Observable<INearbySearchResponse> {
        return this.rq.post(this.apiUrl + 'place/nearby', { next_page_token });
    }

    private _geocode(input: IGeocodeInput): Observable<IGeocodeResult[]> {
        return this.rq.post(this.apiUrl + 'geocode', input).pipe(map((res: IGeocodeResponse) => {
            return res.results;
        }));
    }

    private _reverseGeocode(input: IReverseGeocodeInput): Observable<IGeocodeResult[]> {
        return this.rq.post(this.apiUrl + 'reverse-geocode', input).pipe(map((res: IReverseGeocodeResponse) => {
            return res.results;
        }));
    }
}
