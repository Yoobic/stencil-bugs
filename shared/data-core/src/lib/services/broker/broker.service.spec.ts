import { Injectable, EventEmitter, Injector } from '@angular/core';
import { Requestor } from '../requestor/requestor.service';
import { Models } from '../models/models.service';
import { Googlemaps } from '../googlemaps/googlemaps.service';
import { Workplace } from '../workplace/workplace.service';
import { Session } from '../session/session.service';
import { Files } from '../files/files.service';
import { Box } from '../box/box.service';
import { Config } from '../config/config.service';
import { ResponseObject } from '../../interfaces/response-object/response-object.interface';
import { Location } from '../../interfaces/location/location.interface';
import { Observable, Observer, forkJoin, from, of } from 'rxjs';
import { Network } from '@shared/common';
import { Broker } from './broker.service';

describe('data-core: services: broker', () => {
    //let rq, config, network, session;
    let mockGet = jest.fn<Observable<any>>(() => of(null));
    let mockPost = jest.fn<Observable<any>>(() => of(null));
    let mockPut = jest.fn<Observable<any>>(() => of(null));
    let mockPatch = jest.fn<Observable<any>>(() => of(null));
    let broker;

    beforeEach(() => {
        const mockConfig = jest.fn<Config>(() => ({
            isTestPen: jest.fn(() => false),
            apiUrl: 'api.mock/'
        }));
        let config = new mockConfig();

        const mockNetwork = jest.fn<Network>(() => ({
            isOffline: jest.fn(() => false),
        }));
        let network = new mockNetwork();

        const mockSession = jest.fn<Session>(() => ({
            user: '1111'
        }));
        let session = new mockSession();

        const mockFiles = jest.fn<Files>(() => ({
            isImageFile: jest.fn(() => false)
        }));
        let files = new mockFiles();

        const mockRq = jest.fn<Requestor>(() => ({
            get: mockGet,
            post: mockPost,
            put: mockPut,
            patch: mockPatch
        }));
        let rq = new mockRq();
        broker = new Broker(rq, null, null, null, session, files, null, null, null, network, null, null, config, null);
        jest.clearAllMocks();
    });
    describe('getById', () => {
        it('should get by id', async () => {
            let result;
            broker.getById('collection', '1').subscribe();
            expect(mockGet).toMatchSnapshot();
        });

        it('should be get by a non standard id', () => {
            let result;
            broker.getById('collection', '1', undefined, undefined, 'notid').subscribe();
            expect(mockPost).toMatchSnapshot();
        });
    });

    describe('upsert', () => {
        const ENTITY = {
            title: 'Test',
            background: {
                _downloadURL : 'https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1513348890/dwitbomq9ilccln6702l.png',
                _filename : 'Screen shot 2017-12-15 aÌ€ 15.32.33.png'
            },
            'f84ab24a-d931-4b4a-a7bc-6a167acc22a5': {
                value : 'https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1504262031/tobfp8xpdq8qgpoaxfdv.jpg',
                fieldType : 'photo',
                fieldTitle : 'Photo'
            }
        };
        let result;
        it('should upsert an existing entity', async () => {
            broker.upsert('collection', {_id: 1, ...ENTITY }).subscribe();
            expect(mockPatch).toMatchSnapshot();
        });

        it('should upsert with a new entity', () => {
            broker.upsert('collection', ENTITY).subscribe();
            expect(mockPost).toMatchSnapshot();
        });

        it('should upsert missiondatas', () => {
            broker.upsert('missiondatas', ENTITY).subscribe();
            expect(mockPost).toMatchSnapshot();
        });

        it('should upsert with previous entity', () => {

        });

    });
});