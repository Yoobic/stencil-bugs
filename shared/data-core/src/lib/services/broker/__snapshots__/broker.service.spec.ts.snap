// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-core: services: broker getById should be get by a non standard id 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "api.mock/collection/find",
      Object {
        "filter": Object {
          "limit": 20,
          "order": Array [
            "_lmt DESC",
          ],
          "where": Object {
            "notid": "1",
          },
        },
      },
      null,
      true,
      false,
      null,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Observable {
        "_isScalar": true,
        "_subscribe": [Function],
        "value": null,
      },
    },
  ],
}
`;

exports[`data-core: services: broker getById should get by id 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "api.mock/collection/1?filter=%7B%7D",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Observable {
        "_isScalar": true,
        "_subscribe": [Function],
        "value": null,
      },
    },
  ],
}
`;

exports[`data-core: services: broker upsert should upsert an existing entity 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "api.mock/collection/1",
      Object {
        "_acl": Object {
          "creator": undefined,
          "groups": Object {
            "r": Array [],
            "w": Array [],
          },
        },
        "_id": 1,
        "background": Object {
          "_downloadURL": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1513348890/dwitbomq9ilccln6702l.png",
          "_filename": "Screen shot 2017-12-15 à 15.32.33.png",
        },
        "f84ab24a-d931-4b4a-a7bc-6a167acc22a5": Object {
          "fieldTitle": "Photo",
          "fieldType": "photo",
          "value": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1504262031/tobfp8xpdq8qgpoaxfdv.jpg",
        },
        "title": "Test",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Observable {
        "_isScalar": true,
        "_subscribe": [Function],
        "value": null,
      },
    },
  ],
}
`;

exports[`data-core: services: broker upsert should upsert missiondatas 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "api.mock/missiondatas",
      Object {
        "_acl": Object {
          "creator": undefined,
          "groups": Object {
            "r": Array [],
            "w": Array [],
          },
        },
        "background": Object {
          "_downloadURL": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1513348890/dwitbomq9ilccln6702l.png",
          "_filename": "Screen shot 2017-12-15 à 15.32.33.png",
        },
        "f84ab24a-d931-4b4a-a7bc-6a167acc22a5": Object {
          "fieldTitle": "Photo",
          "fieldType": "photo",
          "value": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1504262031/tobfp8xpdq8qgpoaxfdv.jpg",
        },
        "title": "Test",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Observable {
        "_isScalar": true,
        "_subscribe": [Function],
        "value": null,
      },
    },
  ],
}
`;

exports[`data-core: services: broker upsert should upsert with a new entity 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "api.mock/collection",
      Object {
        "_acl": Object {
          "creator": undefined,
          "groups": Object {
            "r": Array [],
            "w": Array [],
          },
        },
        "background": Object {
          "_downloadURL": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1513348890/dwitbomq9ilccln6702l.png",
          "_filename": "Screen shot 2017-12-15 à 15.32.33.png",
        },
        "f84ab24a-d931-4b4a-a7bc-6a167acc22a5": Object {
          "fieldTitle": "Photo",
          "fieldType": "photo",
          "value": "https://res.cloudinary.com/www-yoobic-com/image/upload/a_exif/v1504262031/tobfp8xpdq8qgpoaxfdv.jpg",
        },
        "title": "Test",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Observable {
        "_isScalar": true,
        "_subscribe": [Function],
        "value": null,
      },
    },
  ],
}
`;
